aws/openid/github/openid.tf

```diff
@@ -0,0 +1,74 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+resource "aws_iam_role" "github_oidc_role" {
+  name = "github-action-${var.name}"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [{
+      Action = "sts:AssumeRoleWithWebIdentity",
+      Effect = "Allow",
+      Principal = {
+        Federated = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/token.actions.githubusercontent.com"
+      },
+      "Condition" : {
+        "StringEquals" : {
+          "token.actions.githubusercontent.com:aud" : "sts.amazonaws.com"
+        }
+        "StringLike" : {
+          "token.actions.githubusercontent.com:sub" : "repo:${var.repo}:*"
+        },
+      }
+    }]
+  })
+
+  inline_policy {
+    name = "ReadWriteS3StateBucket"
+    policy = jsonencode({
+      Version : "2012-10-17",
+      Statement : [
+        {
+          Effect : "Allow",
+          Action : [
+            "s3:GetObject",
+            "s3:ListBucket",
+            "s3:PutObject"
+          ],
+          Resource : [
+            "arn:aws:s3:::${var.name}-tfstate",
```

Comment:
QoL:

Mhh.., might be wrong on this, but it seems that this more or less makes the assumption that s3 state buckets are always suffixed with "-tfstate" which makes this module only usable from the outside if it also uses our "setup_tfstate" module (and if s3 buckets are even used as storage backend).

In order to work accordingly for other state backends or even other s3 state bucket naming schemes users have to specify the permissions themselves, which is fine. However, users will also end up with permissions they are not using, since the "ReadWriteS3StateBucket" and "DynamoDbStateLock" will always be created. Which might be undesireable.

Perhaps we can add something like a "createS3StorageBackendPermissions" flag?